interview-questions.md

```diff
@@ -7,90 +7,111 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. Why would you use the method `super()`?
 
-  Your answer:
+  Your answer: super() is used to gain access to functions in a parent class
 
-  Researched answer:
+  Researched answer: By calling the super() method in the constructor method, we call the parent's constructor method and gets access to the parent's properties and methods.
 
 
 
 2. What is the DOM? What is the virtual DOM?
 
-  Your answer:
+  Your answer: Document Object Model, and virtual DOM sits somewhere between our code and the real DOM, but I'm not really confident what they are
 
-  Researched answer:
+  Researched answer: The DOM is a W3C (World Wide Web Consortium) standard. The DOM defines a standard for accessing documents: "The W3C Document Object Model (DOM) is a platform and language-neutral interface that allows programs and scripts to dynamically access and update the content, structure, and style of a document."
+  The DOM is a W3C (World Wide Web Consortium) standard.
+
+  The DOM defines a standard for accessing documents:
+
+  "The W3C Document Object Model (DOM) is a platform and language-neutral interface that allows programs and scripts to dynamically access and update the content, structure, and style of a document."
+
+  The W3C DOM standard is separated into 3 different parts:
+
+  Core DOM - standard model for all document types
+  XML DOM - standard model for XML documents
+  HTML DOM - standard model for HTML documents
 
 
 
 3. What is a higher-order function?
 
-  Your answer:
+  Your answer: They are functions that take other functions as arguments.
 
-  Researched answer:
+  Researched answer: Higher-order functions are functions that take other functions as arguments or return functions as their results. Taking an other function as an argument is often referred as a callback function, because it is called back by the higher-order function. The term comes from mathematics, where the distinction between functions and other values is taken more seriously. Higher-order functions allow us to abstract over actions, not just values.
 
 
 
 4. What is JSX?
 
-  Your answer:
+  Your answer: Javascript XML, allowing us to write HTML in our Javascript.
 
-  Researched answer:
+  Researched answer: JSX stands for JavaScript XML. JSX allows us to write HTML in React. JSX makes it easier to write and add HTML in React. JSX is a preprocessor step that adds XML syntax to JavaScript. Just like XML, JSX tags have a tag name, attributes, and children. If an attribute value is enclosed in quotes, the value is a string. Otherwise, wrap the value in braces and the value is the enclosed JavaScript expression.
 
 
 
 5. What is React?
 
-  Your answer:
+  Your answer: React is a Javascript library and is component based; created by Facebook.
```

Comment:
Absolutely. I would want to expand on it and mention that its specifically a Library allowing us to create dynamic, component based, reactive front-end applications.